api/v1alpha1/spaceprovisionerconfig_types.go

```diff
@@ -50,10 +53,28 @@ type SpaceProvisionerCapacityThresholds struct {
 	MaxMemoryUtilizationPercent uint `json:"maxMemoryUtilizationPercent,omitempty"`
 }
 
+// ConsumedCapacity describes the capacity of the cluster consumed by the spaces
+// currently provisioned to it.
+type ConsumedCapacity struct {
+	// MemoryUsagePercentPerNodeRole is the percent of the memory used per node role (eg. worker, master)
+	MemoryUsagePercentPerNodeRole map[string]int `json:"memoryUsagePercentPerNode"`
+
+	// SpaceCount is the number of spaces currently deployed to the cluster
+	SpaceCount int `json:"spaceCount"`
```

Comment:
same here - I usually saw that if the parent struct is optional also the fields inside are optional
```suggestion
	// SpaceCount is the number of spaces currently deployed to the cluster
	// +optional
	SpaceCount int `json:"spaceCount,omitempty"`
```

Reply:
The reason for this is that we either have a complete capacity report or none at all. Optional and omitempty only influence the JSON that is marshalled or presented to the user and since this is a status and users are not supposed to edit it, I wouldn't find it useful to provide partial capacity reports or as a user to not see "spaceCount: 0" in the status.

Reply:
nope, we will assign the consumed capacity as a whole, not the individual fields.